core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -572,6 +693,38 @@ public ConnectionDetails getConnectionDetails( String name ) {
     return detailsByName.get( name );
   }
 
+  /**
+   * Gets a connection given its name, casting it to the type parameter.
+   *
+   * @param name The connection name
+   * @return The named connection details
+   */
+  @SuppressWarnings( "unchecked" )
+  @Nullable
+  public <T extends ConnectionDetails> T getDetails( @Nullable String name ) {
+    return (T) getConnectionDetails( name );
+  }
+
+  /**
+   * Gets the details of a connection, given its name, casting it to the type parameter,
+   * and throwing if it does not exist.
+   *
+   * @param name The connection name
+   * @return The named connection details
+   * @throws KettleException When a connection with the given name is not defined.
+   */
+  @NonNull
+  public <T extends ConnectionDetails> T getExistingDetails( @Nullable String name )
+    throws KettleException {
+
+    T details = getDetails( name );
+    if ( details == null ) {
+      throw new KettleException( String.format( "Undefined connection '%s'.", name ) );
+    }
+
+    return details;
+  }
+
```

Comment:
These new methods are sugar that perform casting of the result to the caller's result type. This removes the need for a lot of helper methods or code with unchecked warnings  proliferating out there. The variant which assumes existence also simplifies usage a lot.